src/seedu/addressbook/commands/AddCommand.java

```diff
@@ -60,10 +61,16 @@ public ReadOnlyPerson getPerson() {
     public CommandResult execute() {
         try {
             addressBook.addPerson(toAdd);
-            return new CommandResult(String.format(MESSAGE_SUCCESS, toAdd));
+            List<ReadOnlyPerson> allPersons = addressBook.getAllPersons().immutableListView();
+            return new CommandResult(String.format(MESSAGE_SUCCESS, toAdd), allPersons);
         } catch (UniquePersonList.DuplicatePersonException dpe) {
             return new CommandResult(MESSAGE_DUPLICATE_PERSON);
         }
     }
 
+    @Override
+    public boolean isMutating(String userCommandText) {
```

Comment:
There is actually no polymorphism as the subclass simply calls the super class method. A better way to specify the  behaviors of different commands would be to remove the default implementation in super class, and having each sub class returning True/False.